### PR TITLE
[Component with its own class name as service type] LPS-197867 Refactor components from dynamic-data-mapping modules

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-197867

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:forms

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
@MrJohn1911 regression found:

```
[beanshell]      [exec] com.liferay.dynamic.data.mapping.service.test.DDMTemplateLocalServiceTest > testSearchByNameAndDescription1 FAILED
[beanshell]      [exec]     java.lang.RuntimeException: Unable to get class name from id 0
[beanshell]      [exec]     	at com.liferay.portal.util.PortalImpl.getClassName(PortalImpl.java:1692)
[beanshell]      [exec]     	at com.liferay.dynamic.data.mapping.internal.security.permission.support.DDMPermissionSupportImpl.getTemplateModelResourceName(DDMPermissionSupportImpl.java:176)
[beanshell]      [exec]     	at com.liferay.dynamic.data.mapping.service.impl.DDMTemplateLocalServiceImpl.search(DDMTemplateLocalServiceImpl.java:1261)
[beanshell]      [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[beanshell]      [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[beanshell]      [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[beanshell]      [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
[beanshell]      [exec]     	at com.liferay.portal.spring.aop.AopMethodInvocationImpl.proceed(AopMethodInvocationImpl.java:41)
[beanshell]      [exec]     	at com.liferay.portal.spring.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:60)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-11/test-1-28/test-portal-acceptance-pullrequest(master)/8308/modules-integration-mysql57-jdk8/0/3/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:forms

**Comment by catarineaguiar** [comment]
no regressions found. 

@MrJohn1911 can you resolve conflicts and re-run only sf and relevant, please?
Thanks!!

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by pyoo47** [comment]
@MrJohn1911 I had to abort the `relevant` test because it froze in CI

**Comment by pyoo47** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java [hunk 1]
```diff
@@ -1039,10 +1039,24 @@ public List<DDMTemplate> search(
 		int status, int start, int end,
 		OrderByComparator<DDMTemplate> orderByComparator) {
 
+		String templateModelResourceClassName = null;
```
Comment (rafaprax) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java:1042]: Hey @MrJohn1911, do not create this temp variable. You need to apply the same pattern we have into the `DDMStructureLocalServiceImpl`. Thx

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java (hunk 1 above)
Comment (MrJohn1911) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMTemplateLocalServiceImpl.java:1042]: Hey @rafaprax I tried to do it before, but if I inline this variable in the method and wrap it with the try/catch block, the searchContext is never created, changing the logic that were applied before my changes.

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Pull request closed without merging**